agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -222,12 +221,54 @@ func GetTaskProtectionHandler(state dockerstate.TaskEngineState,
 			return
 		}
 
-		// TODO: Call ECS
 		logger.Info("GetTaskProtection endpoint was called", logger.Fields{
 			loggerfield.Cluster: cluster,
 			loggerfield.TaskARN: task.Arn,
 		})
-		writeJSONResponse(w, http.StatusOK, "Ok", getTaskProtectionRequestType)
+
+		ecsClient, responseCode, err := factory.newTaskProtectionClient(credentialsManager, task)
+		if err != nil {
+			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
+		response, err := ecsClient.GetTaskProtection(&ecs.GetTaskProtectionInput{
+			Cluster: aws.String(cluster),
+			Tasks:   aws.StringSlice([]string{task.Arn}),
+		})
+
+		if err != nil {
+			exceptionType, statusCode := getExceptionTypeAndStatusCode(err)
+			logger.Error("ECS throws an exception.", logger.Fields{
```

Comment:
Should the error message be something like "Error getting task protection data: %v - %v", exception type, err

Reply:
I will change to something like this: 
```
Get an exception when calling UpdateTaskProtection.
```